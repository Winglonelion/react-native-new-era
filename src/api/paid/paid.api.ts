import axios from 'axios';
import {
  PAID_HISTORY_OVERVIEW_ENDPOINT,
  PAID_OVERVIEW_ENDPOINT,
} from '../api.endpoint';
import {
  GetPaidHistoryOverViewParameters,
  GetPaidHistoryOverViewResponse,
  GetPaidOverViewParameters,
  GetPaidOverViewResponse,
} from './paid.api.types';

export function getPaidOverViewAPI({ userId }: GetPaidOverViewParameters) {
  return axios.post<GetPaidOverViewResponse>(PAID_OVERVIEW_ENDPOINT, {
    user_id: userId,
  });
}

export function getPaidHistoryOverViewAPI({
  userId,
}: GetPaidHistoryOverViewParameters) {
  return axios.post<GetPaidHistoryOverViewResponse>(
    PAID_HISTORY_OVERVIEW_ENDPOINT,
    {
      user_id: userId,
    },
  );
}
