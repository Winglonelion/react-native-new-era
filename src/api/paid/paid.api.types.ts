export type GetPaidOverViewParameters = {
  userId: string;
};

export type GetPaidOverViewResponse = {
  take_home: number;
  gross: number;
  hours: number;
  rate: number;
  regular: number;
  over_time: number;
  detail: { type: string; value: number }[];
};

export type GetPaidHistoryOverViewParameters = {
  userId: string;
};

export type GetPaidHistoryOverViewResponse = {
  date: string;
  take_home: number;
  gross: number;
  hours: number;
}[];
