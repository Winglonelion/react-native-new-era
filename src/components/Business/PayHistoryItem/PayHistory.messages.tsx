const messages = {
  take_home: 'Take Home',
  hours: 'Hours',
  gross: 'Gross',
};

export default messages;
