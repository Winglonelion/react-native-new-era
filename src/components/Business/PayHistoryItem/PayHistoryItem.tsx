import React, { memo } from 'react';
import { Image, StyleSheet, View } from 'react-native';

import Colors from 'theme/colors';
import Space from 'components/Base/Space/Space';
import DateTitle from 'components/Common/DateTitle';
import TitleText from 'components/Base/Text/TitleText';
import MoneyValue from 'components/Base/Text/MoneyValue/';
import ContentText from 'components/Base/Text/ContentText';
import ContentLoaderLine from 'components/Base/ContentLoader/BuildIn/ContentLoaderLine';
import messages from './PayHistoryItem.messages';

export interface PropTypes {
  item?: {
    date: string;
    take_home: number;
    hours: number;
    gross: number;
  };
}

const PayHistoryItem: React.FC<PropTypes> = memo(
  ({ item = payHistoryItemDefaultProps.item }) => {
    return (
      <View style={styles.container}>
        <View style={styles.contentView}>
          <ContentLoaderLine
            width={73}
            height={18}
            ready={typeof item.date === 'string'}>
            <DateTitle level="h5" date={new Date(item?.date ?? '')} />
          </ContentLoaderLine>
          <Space height={8} />
          <View style={styles.dataRow}>
            {/* take home  */}
            <View style={styles.dataRowItem}>
              <TitleText level="h5" color={Colors.corban}>
                {messages.take_home}
              </TitleText>
              <ContentLoaderLine
                width={80}
                height={22}
                ready={typeof item.take_home === 'number'}>
                <ContentText size={16}>
                  <MoneyValue value={item.take_home} />
                </ContentText>
              </ContentLoaderLine>
            </View>
            {/* hours */}
            <View style={styles.dataRowItem}>
              <TitleText level="h5" color={Colors.corban}>
                {messages.hours}
              </TitleText>
              <ContentLoaderLine
                width={80}
                height={22}
                ready={typeof item.hours === 'number'}>
                <ContentText size={16}>{item.hours?.toFixed(2)}</ContentText>
              </ContentLoaderLine>
            </View>
            {/* gross */}
            <View style={styles.dataRowItem}>
              <TitleText level="h5" color={Colors.corban}>
                {messages.gross}
              </TitleText>
              <ContentLoaderLine
                width={80}
                height={22}
                ready={typeof item.gross === 'number'}>
                <ContentText size={16}>
                  <MoneyValue value={item.gross} />
                </ContentText>
              </ContentLoaderLine>
            </View>
          </View>
        </View>
        <Image
          source={require('images/chevron_right_corban.png')}
          style={styles.iconImage}
        />
      </View>
    );
  },
);

const payHistoryItemDefaultProps = {
  item: {
    date: undefined,
    take_home: undefined,
    hours: undefined,
    gross: undefined,
  },
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    width: '100%',
    alignItems: 'center',
    marginVertical: 18,
  },
  contentView: {
    flex: 1,
    alignItems: 'flex-start',
    // paddingHorizontal: 16,
  },
  dataRow: {
    flexDirection: 'row',
    width: '95%',
    justifyContent: 'space-between',
  },
  dataRowItem: {
    flex: 1,
  },
  iconImage: {
    width: 24,
    height: 24,
    resizeMode: 'contain',
  },
});

export default PayHistoryItem;
