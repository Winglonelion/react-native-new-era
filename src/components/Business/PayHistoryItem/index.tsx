export { default as default } from './PayHistoryItem';
