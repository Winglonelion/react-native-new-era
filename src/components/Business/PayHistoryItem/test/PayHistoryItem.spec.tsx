import React from 'react';
import { shallow } from 'enzyme';
import PayHistoryItem, { PropTypes } from '../PayHistoryItem';
import { paid_history_overview } from 'mock/networking/paid.mock';

const setup = (propOverrides: PropTypes = {} as PropTypes) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<PayHistoryItem {...props} />);
  return {
    props,
    wrapper,
  };
};

describe('PayHistory check snapshot', () => {
  test('should match snapshot with props', () => {
    const { wrapper } = setup({
      item: paid_history_overview[0],
    });
    expect(wrapper).toMatchSnapshot();
  });
  test('should match snapshot without props and loading', () => {
    const { wrapper } = setup({ item: undefined });
    expect(wrapper).toMatchSnapshot();
  });
});
