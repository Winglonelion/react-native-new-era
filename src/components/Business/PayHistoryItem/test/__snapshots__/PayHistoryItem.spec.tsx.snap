// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayHistory check snapshot should match snapshot with props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginVertical": 18,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
      }
    }
  >
    <Memo()
      height={18}
      ready={true}
      width={73}
    >
      <Memo()
        date={2021-01-31T00:00:00.000Z}
        level="h5"
      />
    </Memo()>
    <Memo()
      height={8}
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "95%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Memo()
          color="#8EADAD"
          level="h5"
        >
          Take Home
        </Memo()>
        <Memo()
          height={22}
          ready={true}
          width={80}
        >
          <Memo()
            size={16}
          >
            <Memo()
              value={2200.01}
            />
          </Memo()>
        </Memo()>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Memo()
          color="#8EADAD"
          level="h5"
        >
          Hours
        </Memo()>
        <Memo()
          height={22}
          ready={true}
          width={80}
        >
          <Memo()
            size={16}
          >
            180.00
          </Memo()>
        </Memo()>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Memo()
          color="#8EADAD"
          level="h5"
        >
          Gross
        </Memo()>
        <Memo()
          height={22}
          ready={true}
          width={80}
        >
          <Memo()
            size={16}
          >
            <Memo()
              value={2400}
            />
          </Memo()>
        </Memo()>
      </View>
    </View>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../src/images/chevron_right_corban.png",
      }
    }
    style={
      Object {
        "height": 24,
        "resizeMode": "contain",
        "width": 24,
      }
    }
  />
</View>
`;

exports[`PayHistory check snapshot should match snapshot without props and loading 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginVertical": 18,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
      }
    }
  >
    <Memo()
      height={18}
      ready={false}
      width={73}
    >
      <Memo()
        date={Date { NaN }}
        level="h5"
      />
    </Memo()>
    <Memo()
      height={8}
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "95%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Memo()
          color="#8EADAD"
          level="h5"
        >
          Take Home
        </Memo()>
        <Memo()
          height={22}
          ready={false}
          width={80}
        >
          <Memo()
            size={16}
          >
            <Memo() />
          </Memo()>
        </Memo()>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Memo()
          color="#8EADAD"
          level="h5"
        >
          Hours
        </Memo()>
        <Memo()
          height={22}
          ready={false}
          width={80}
        >
          <Memo()
            size={16}
          />
        </Memo()>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Memo()
          color="#8EADAD"
          level="h5"
        >
          Gross
        </Memo()>
        <Memo()
          height={22}
          ready={false}
          width={80}
        >
          <Memo()
            size={16}
          >
            <Memo() />
          </Memo()>
        </Memo()>
      </View>
    </View>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../src/images/chevron_right_corban.png",
      }
    }
    style={
      Object {
        "height": 24,
        "resizeMode": "contain",
        "width": 24,
      }
    }
  />
</View>
`;
